src/napari_spatialdata/_utils.py

```diff
@@ -221,3 +203,272 @@ def _points_inside_triangles(points: NDArrayA, triangles: NDArrayA) -> NDArrayA:
+    into c x y x x and flags as suitable for rgb visualization.
+
+    Parameters
+    ----------
+    element: SpatialImage | MultiScaleSpatialImage
+        Element in sdata.images
+
+    Returns
+    -------
+    new_raster: DataArray
+        The image in shape of c x y x x.
+    rgb: bool
+        Flag indicating suitability for rgb(a) visualization
+    """
+    axes = get_axes_names(element)
+
+    if "c" in axes:
+        assert axes.index("c") == 0
+        if isinstance(element, SpatialImage):
+            n_channels = element.shape[0]
+        elif isinstance(element, MultiscaleSpatialImage):
+            v = element["scale0"].values()
+            assert len(v) == 1
+            n_channels = v.__iter__().__next__().shape[0]
+        else:
+            raise TypeError(f"Unsupported type for images or labels: {type(element)}")
+    else:
+        n_channels = 0
+
+    if n_channels in [3, 4]:
+        rgb = True
+        new_raster = element.transpose("y", "x", "c")
+    else:
+        rgb = False
+        new_raster = element
+
+    # TODO: after we call .transpose() on a MultiscaleSpatialImage object we get a DataTree object. We should look at
+    # this better and either cast somehow back to MultiscaleSpatialImage or fix/report this
+    if isinstance(new_raster, (MultiscaleSpatialImage, DataTree)):
+        list_of_xdata = []
+        for k in new_raster:
+            v = new_raster[k].values()
+            assert len(v) == 1
+            xdata = v.__iter__().__next__()
+            list_of_xdata.append(xdata)
+        new_raster = list_of_xdata
+
+    return new_raster, rgb
+
+
+def _init_colors_for_obs(adata: AnnData) -> AnnData:
+    from scanpy.plotting._utils import _set_colors_for_categorical_obs
+
+    # quick and dirty to set the colors for all the categorical dtype so that if we subset the table the
+    # colors are consistent
+
+    for key in adata.obs:
+        if is_categorical_dtype(adata.obs[key]) and f"{key}_colors" not in adata.uns:
+            _set_colors_for_categorical_obs(adata, key, palette="tab20")
+    return adata
+
+
+def points_to_anndata(points_element: DaskDataFrame, points: NDArrayA, dims: tuple[str]) -> AnnData | None:
+    annotations_columns = list(set(points_element.columns.to_list()).difference(dims))
+    if len(annotations_columns) > 0:
+        df = points_element[annotations_columns].compute()
+        annotation = AnnData(shape=(len(points), 0), obs=df, obsm={"spatial": points})
+        return _init_colors_for_obs(annotation)
+    return None
+
+
+def _get_mapping_info(annotation_table: AnnData) -> tuple[str, str, str]:
+    regions = annotation_table.uns["spatialdata_attrs"]["region"]
+    regions_key = annotation_table.uns["spatialdata_attrs"]["region_key"]
+    instance_key = annotation_table.uns["spatialdata_attrs"]["instance_key"]
+    return regions, regions_key, instance_key
+
+
+def _find_annotation_for_labels(
```

Comment:
it's unclear why this is needed, in the implementation from main there shouldn't be need for this. Is it possible that this is something you ported over from luca's branch and now it's here for legacy reason?